CollectorUtils/UpdateGNSSSMetadata.py

```diff
@@ -0,0 +1,293 @@
+
+
+# Parse Command-line arguments
+def parseArguments():
+    global args_parser
+    parser = argparse.ArgumentParser(
+        usage='Add/Update GNSS metadate fields')
+    parser.add_argument('-u', '--username', required=True, type=str, help='Organization username')
+    parser.add_argument('-p', '--password', required=True, type=str, help='Organization password')
+    parser.add_argument('-url', '--url', required=True, type=str, help='Organization url')
+    parser.add_argument('-r', '--remove', default=False, type=bool,
+                        help='Set True if GNSS metadata fields need to be removed')
+    parser.add_argument('itemId', type=str, nargs="+", help='Search string')
+    args_parser = parser.parse_args()
+
+
+# Generate Token based on username and password
+def generateToken():
+    global token
+
+    datapayload = bytes(urllib.parse.urlencode( \
+        {'username': args_parser.username, 'password': args_parser.password, 'referer': args_parser.url, 'f': 'json',
+         'expiration': 60}), 'utf-8')
+    token_response = urllib.request.urlopen(urllib.parse.urljoin(args_parser.url, '/sharing/rest/GenerateToken?'),
+                                            datapayload).read()
+    token = json.loads(token_response)['token']
+    if not token:
+        print('Failed to generate Token .. ')
+        sys.exit(-2)
+
+
+# Search for a item id and add GNSS Metadata fields
+def searchItems_addGNSSMetadataFields():
+    global featureServiceUrl
+
+    # Search ItemIds
+    gis = GIS(args_parser.url, args_parser.username, args_parser.password)
+    itemId = args_parser.itemId
+
+    try:
+        # Iterate through each ItemId and update the domain values
+        for id in itemId:
+            featurelayerItem = gis.content.search(id)
+            featureServiceUrl = featurelayerItem[0].layers[0].url
+
+            # Construct the Payload
+            datapayload = bytes(urllib.parse.urlencode({'token': token}), 'utf-8')
+
+            featureLayerResponse = urllib.request.urlopen(urllib.parse.urljoin(featureServiceUrl, '?&f=pjson'),
+                                                          datapayload).read()
+            featureLayerJson = json.loads(featureLayerResponse)
+
+            # Extract fields from Feature layer service definition
+            featureLayerFields = featureLayerJson['fields']
+
+            # New fields which need to be added
+            gnssMetadataFields = {'fields': []}
+
+            # Operations list - Add, Update or delete GNSS Metadata fields.
+            operations = []
+
+            # Add/Update GNSS Metadata fields
+            if not args_parser.remove:                
+
+                # ESRIGNSS_FIXDATETIME
+                fixTimeField = [field for field in featureLayerFields if field['name'] == 'ESRIGNSS_FIXDATETIME']
+
+                if not fixTimeField:
+                    gnssMetadataFields['fields'].append({'name': 'ESRIGNSS_FIXDATETIME', 'type': 'esriFieldTypeDate', \
```

Comment:
I would put each field attribute on it's own line so it's easier to read. Something like: 
```python
{
    'name': 'ESRIGNSS_FIXDATETIME',
    'type': 'esriFieldTypeDate',
    'alias': 'Fix Time',
    'sqlType': 'sqlTypeOther',
    'length': 0,
    'nullable': True,
    'editable' True,
    'domain': None,
    'defaultValue' None
}

I realize it makes the script longer but it's more "pythonic" and easier to see what each field is.